packages/editor-ui/src/components/WorkflowCard.vue

```diff
@@ -1,66 +1,240 @@
+				id: '',
+				createdAt: '',
+				updatedAt: '',
+				active: false,
+				connections: {},
+				nodes: [],
+				name: '',
+				sharedWith: [],
+				ownedBy: {} as IUser,
+			}),
+		},
+		readonly: {
 			type: Boolean,
+			default: false,
+		},
+	},
+	computed: {
+		currentUser (): IUser {
+			return this.$store.getters['users/currentUser'];
 		},
-		title: {
-			type: String,
+		areTagsEnabled(): boolean {
+			return this.$store.getters['settings/areTagsEnabled'];
+		},
+		credentialPermissions(): IPermissions {
+			return getWorkflowPermissions(this.currentUser, this.data, this.$store);
+		},
+		actions(): Array<{ label: string; value: string; }> {
+			return [
+				{
+					label: this.$locale.baseText('workflows.item.open'),
+					value: WORKFLOW_LIST_ITEM_ACTIONS.OPEN,
+				},
+			].concat(this.credentialPermissions.delete ? [{
+				label: this.$locale.baseText('workflows.item.delete'),
+				value: WORKFLOW_LIST_ITEM_ACTIONS.DELETE,
+			}]: []);
+		},
+		formattedCreatedAtDate(): string {
+			const currentYear = new Date().getFullYear();
+
+			return dateformat(this.data.createdAt, `d mmmm${this.data.createdAt.startsWith(currentYear) ? '' : ', yyyy'}`);
+		},
+	},
+	methods: {
+		async onClick(event?: PointerEvent) {
+			if (event) {
+				if ((this.$refs.activator as Vue)?.$el.contains(event.target as HTMLElement)) {
+					return;
+				}
+
+				if (event.metaKey || event.ctrlKey) {
+					const route = this.$router.resolve({name: VIEWS.WORKFLOW, params: { name: this.data.id }});
+					window.open(route.href, '_blank');
+
+					return;
+				}
+			}
+
+			this.$router.push({
+				name: VIEWS.WORKFLOW,
+				params: { name: this.data.id },
+			});
+		},
+		async onAction(action: string) {
+			if (action === WORKFLOW_LIST_ITEM_ACTIONS.OPEN) {
+				this.onClick();
+			} else if (action === WORKFLOW_LIST_ITEM_ACTIONS.DELETE) {
+				const deleteConfirmed = await this.confirmMessage(
+					this.$locale.baseText(
+						'mainSidebar.confirmMessage.workflowDelete.message',
+						{ interpolate: { workflowName: this.data.name } },
+					),
+					this.$locale.baseText('mainSidebar.confirmMessage.workflowDelete.headline'),
+					'warning',
+					this.$locale.baseText('mainSidebar.confirmMessage.workflowDelete.confirmButtonText'),
+					this.$locale.baseText('mainSidebar.confirmMessage.workflowDelete.cancelButtonText'),
+				);
+
+				if (deleteConfirmed === false) {
+					return;
+				}
+
+				try {
+					await this.restApi().deleteWorkflow(this.data.id);
+					this.$store.commit('deleteWorkflow', this.data.id);
+				} catch (error) {
+					this.$showError(
+						error,
+						this.$locale.baseText('mainSidebar.showError.stopExecution.title'),
+					);
+					return;
+				}
+
+				// Reset tab title since workflow is deleted.
+				this.$showMessage({
+					title: this.$locale.baseText('mainSidebar.showMessage.handleSelect1.title'),
+					type: 'success',
+				});
+			}
+		},
+		onActiveChange (data: { id: string, active: boolean }) {
```

Comment:
we should move this to workflow activator level.. that's where we are making the api request now

Reply:
Wasn't entirely sure how it works. Thanks for pointing that out! Fixed now.